src/uci_handler.h

```diff
@@ -175,6 +177,10 @@ class UciHandler {
 
     static bool handleGo(std::string_view args)
     {
+        if (!s_searchersInitialized) {
+            s_evaluator.initializeSearchers(s_numThreads);
+            s_searchersInitialized = true;
+        }
```

Comment:
i don't think that this is the consumer's job.

I think that it would be neat to simply pass amount of threads to the `getBestMove` and it will handle it for us internally :)